### PR TITLE
Storage config generation

### PR DESCRIPTION
**Pull request by illusional**
- Support generating `storage` config for each dataset
- Ensure it supports the transitive dependency (except for reference)
  - This means it needs to be released in the correct order.

Example: `gcp-tob-wgs-test.toml`:

```toml
[storage.default]
default = "gs://cpg-tob-wgs-test"
web = "gs://cpg-tob-wgs-test-web"
analysis = "gs://cpg-tob-wgs-test-analysis"
tmp = "gs://cpg-tob-wgs-test-tmp"

[storage.thousand-genomes]
default = "gs://cpg-thousand-genomes-test"
web = "gs://cpg-thousand-genomes-test-web"
analysis = "gs://cpg-thousand-genomes-test-analysis"
tmp = "gs://cpg-thousand-genomes-test-tmp"

[storage.tob-wgs]
default = "gs://cpg-tob-wgs-test"
web = "gs://cpg-tob-wgs-test-web"
analysis = "gs://cpg-tob-wgs-test-analysis"
tmp = "gs://cpg-tob-wgs-test-tmp"
```
 
 Example: `gcp-tob-wgs-main.toml`
 
 ```toml
[storage.default]
default = "gs://cpg-tob-wgs-main"
web = "gs://cpg-tob-wgs-main-web"
analysis = "gs://cpg-tob-wgs-main-analysis"
tmp = "gs://cpg-tob-wgs-main-tmp"

[storage.default.test]
default = "gs://cpg-tob-wgs-test"
web = "gs://cpg-tob-wgs-test-web"
analysis = "gs://cpg-tob-wgs-test-analysis"
tmp = "gs://cpg-tob-wgs-test-tmp"

[storage.thousand-genomes]
default = "gs://cpg-thousand-genomes-main"
web = "gs://cpg-thousand-genomes-main-web"
analysis = "gs://cpg-thousand-genomes-main-analysis"
tmp = "gs://cpg-thousand-genomes-main-tmp"

[storage.thousand-genomes.test]
default = "gs://cpg-thousand-genomes-test"
web = "gs://cpg-thousand-genomes-test-web"
analysis = "gs://cpg-thousand-genomes-test-analysis"
tmp = "gs://cpg-thousand-genomes-test-tmp"

[storage.tob-wgs]
default = "gs://cpg-tob-wgs-main"
web = "gs://cpg-tob-wgs-main-web"
analysis = "gs://cpg-tob-wgs-main-analysis"
tmp = "gs://cpg-tob-wgs-main-tmp"

[storage.tob-wgs.test]
default = "gs://cpg-tob-wgs-test"
web = "gs://cpg-tob-wgs-test-web"
analysis = "gs://cpg-tob-wgs-test-analysis"
tmp = "gs://cpg-tob-wgs-test-tmp"
```

### CODE REVIEW

**Comment by lgruen** [comment]
> I just don't have an easy way to tell which place to write configs.

Aren't you writing to `gs://cpg-config/templates/storage/` at the moment, which seems a good place for this?

**Review by MattWellie — APPROVED** [review]
LGTM

**Review by illusional — COMMENTED** [review]
This pull-request is ready IMO

**Review by vladsavelyev — APPROVED** [review]
A few comments

File: cpg_infra/abstraction/azure.py [hunk 1]
```diff
@@ -91,6 +91,9 @@ def storage_account(self):
             sku=az.storage.SkuArgs(name='Standard_LRS'),
         )
 
+    def bucket_output_path(self, bucket):
+        return pulumi.Output.concat('hail-az://', self.storage_account.name, '/', bucket.name)
```
[the remark this answers is not in the document]
Comment (illusional) [reply cpg_infra/abstraction/azure.py]: The container name is the bucket name (container is not unique, so prefix with storage_account_name. It's just what the abstraction calls them, as that what GCP / AWS does, and we don't want to think too much about it.

**Pull request merged**